### PR TITLE
feat: add distinctField for RS API

### PR DESCRIPTION
**Pull request by bietkul**

### CODE REVIEW

File: content/docs/search/reactivesearch-api/reference.md [hunk 1]
```diff
@@ -537,6 +537,62 @@ The following example uses all three functions (`saturation`, `log` and `sigmoid
...
+    ]
+}
+```
+
+### distinctFieldConfig
+This property allows expanding each collapsed top hits with the inner_hits option.
+
+| Type     | Applicable on query of type | Required |
+| ------   | --------------------------- | -------- |
+| `string` | `all`                       | false    |
```
Comment (siddharthlatest) [inline content/docs/search/reactivesearch-api/reference.md]: @bietkul This is an object type, not string.

File: content/docs/search/reactivesearch-api/reference.md [hunk 2]
```diff
@@ -537,6 +537,62 @@ The following example uses all three functions (`saturation`, `log` and `sigmoid
 }
 ```
 
+### distinctField
+You can use the `distinctField` parameter to collapse search results based on field values. The collapsing is done by selecting only the top sorted document per collapse key. You can read more about it [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/collapse-search-results.html).
```
Comment (siddharthlatest) [inline content/docs/search/reactivesearch-api/reference.md]:
```suggestion
This property returns only the distinct value documents for the specified field. It is equivalent to the `DISTINCT` clause in SQL. It internally uses the collapse feature of Elasticsearch. You can read more about it over [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/collapse-search-results.html).
```

File: content/docs/search/reactivesearch-api/reference.md [hunk 3]
```diff
@@ -537,6 +537,62 @@ The following example uses all three functions (`saturation`, `log` and `sigmoid
...
+                "product_name"
+            ],
+            "distinctField": "brand.keyword",
+        }
+    ]
+}
+```
+
+### distinctFieldConfig
+This property allows expanding each collapsed top hits with the inner_hits option.
```
Comment (siddharthlatest) [inline content/docs/search/reactivesearch-api/reference.md]:
```suggestion
This property allows specifying additional options to the `distinctField` property. Using the allowed DSL, one can specify how to return K distinct values (default value of K=1), sort them by a specific order, or return a second level of distinct values. `distinctFieldConfig` object corresponds to the `inner_hits` key's DSL. You can read more about it over [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/collapse-search-results.html).
```

File: content/docs/search/reactivesearch-api/reference.md [hunk 4]
```diff
@@ -537,6 +537,62 @@ The following example uses all three functions (`saturation`, `log` and `sigmoid
...
+
+| Type     | Applicable on query of type | Required |
+| ------   | --------------------------- | -------- |
+| `string` | `all`                       | false    |
```
Comment (bietkul) [inline content/docs/search/reactivesearch-api/reference.md]:
```suggestion
| `object` | `all`                       | false    |
```

**Pull request merged**